code/src/java/pcgen/core/PlayerCharacter.java

```diff
@@ -9756,19 +9719,9 @@ public int getKnownSpellCountForLevel(CDOMList<Spell> list, int level)
 	public List<Spell> getAllSpellsInLists(List<? extends CDOMList<Spell>> spellLists)
 	{
 		List<Spell> spellList = new ArrayList<>();
-		for (CDOMList<Spell> list : availSpellFacet.getScopes1(id))
-		{
-			if (spellLists.contains(list))
-			{
-				for (int lvl : availSpellFacet.getScopes2(id, list))
-				{
-					for (Spell spell : availSpellFacet.getSet(id, list, lvl))
-					{
-						spellList.add(spell);
-					}
-				}
-			}
-		}
+		availSpellFacet.getScopes1(id).stream().filter(spellLists::contains).forEach(list ->
```

Comment:
I think this needs a developer list discussion.  I don't find the compact version of this easier to comprehend